search/search_test.go

```diff
@@ -625,6 +625,103 @@ func TestPutBadStatus(t *testing.T) {
 	}
 }
 
+func TestPutMultiNilIDSlice(t *testing.T) {
+	index, err := Open("Doc")
+	if err != nil {
+		t.Fatalf("err from Open: %v", err)
+	}
+
+	c := aetesting.FakeSingleContext(t, "search", "IndexDocument", func(in *pb.IndexDocumentRequest, out *pb.IndexDocumentResponse) error {
+		if len(in.Params.GetDocument()) < 1 {
+			return fmt.Errorf("got %v, want at least 1 document", in)
+		}
+		got, want := in.Params.Document[0].GetOrderId(), int32(time.Since(orderIDEpoch).Seconds())
+		if d := got - want; -5 > d || d > 5 {
+			return fmt.Errorf("got OrderId %d, want near %d", got, want)
+		}
+		*out = pb.IndexDocumentResponse{
+			Status: []*pb.RequestStatus{
+				{Code: pb.SearchServiceError_OK.Enum()},
+			},
+			DocId: []string{
+				"doc_id",
+			},
+		}
+		return nil
+	})
+
+	if _, err := index.PutMulti(c, nil, []interface{}{&searchFields}); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestPutMultiError(t *testing.T) {
+	index, err := Open("Doc")
+	if err != nil {
+		t.Fatalf("err from Open: %v", err)
+	}
+
+	c := aetesting.FakeSingleContext(t, "search", "IndexDocument", func(in *pb.IndexDocumentRequest, out *pb.IndexDocumentResponse) error {
+		*out = pb.IndexDocumentResponse{
+			Status: []*pb.RequestStatus{
+				{Code: pb.SearchServiceError_OK.Enum()},
+				{Code: pb.SearchServiceError_PERMISSION_DENIED.Enum(), ErrorDetail: proto.String("foo")},
+			},
+			DocId: []string{
+				"id1",
+				"",
+			},
+		}
+		return nil
+	})
+
+	switch _, err := index.PutMulti(c, nil, []interface{}{&searchFields, &searchFields}); {
+	case err == nil:
+		t.Fatalf("got nil, want error")
+	case err.Error() != "search: PERMISSION_DENIED: foo":
```

Comment:
should we be checking the exact error text? seems like a flaky test waiting to happen.